### ISSUE TITLE
BE: Audit test coverage for Forms and Events CMS products

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#14185]
### User story

**AS A** PW BE engineer
**I WANT** to know what test coverage exists for products
**SO THAT** we can protect them during changes, e.g. D10 upgrade.

### Engineering notes / background

See #13990 for notes and thinking driving this ticket.

### Quality / testing notes

- Test types to consider: Cypress (slow), kernel (fast), unit (fastest)

## Acceptance criteria

- [ ] Looked at products for gaps in test coverage:
  - [ ] Forms
  - [ ] Events
- [ ] Ticket written recommending test types, use cases, etc., for products listed above

**Comment by dsasser** [comment]
# Mid-sprint status update 7/11/23

This ticket is at risk of not closing this sprint due to pulling in the [D10 testing ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/13990) (a 13 pointer).

**Comment by jilladams** [comment]
@dsasser I'm going to move this back to Stretch, given that we don't expect it to begin in sprint, and we can pull it into Sprint 89 if we wind up with capacity.

**Comment by jilladams** [comment]
From 89 planning, Daniel notes: 
* We think Events are well covered
* Forms: not that much going on in Drupal under the hood. The migration is the most complex piece. 
* CLPs: not well tested at all
* D10 cypress tests are not passing tests, @edmund-dunn is actively triaging. We may need to confirm scope of what we may need to cover, when he feels done. It's probably not wise to dig in on adding coverage

**Comment by dsasser** [comment]
## Forms

- Integration: no tests but there is little to no value in doing so. There are no major custom code manipulating the content type behavior. There are many content types in Drupal that lack integration tests, so this is not a one-off.
- PHPunit: test exists, and covers the forms migration process
- Migration tests exist. scripts/tasks/VACMS-10735-Forms-CSV-migration-validation.php

## Events

- Integration: cypress covers well the event product including timezone (include the Phoenix tz issue), conditional field display, event series modal and more.
- PHPunit: generic workflow, event subscriber, and other tests cover areas that sprawl over other products. No distinct event unit tests, but all functionality is covered by integration test (where is is more suitable to run these tests).

====
In summary, forms and events are well covered. I don't see any obvious gaps.
